src/tests/nativeaot/SmokeTests/ComWrappers/ComWrappers.cs

```diff
@@ -0,0 +1,171 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if MULTIMODULE_BUILD && !DEBUG
+// Some tests won't work if we're using optimizing codegen, but scanner doesn't run.
+// This currently happens in optimized multi-obj builds.
+#define OPTIMIZED_MODE_WITHOUT_SCANNER
+#endif
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+
+// Name of namespace matches the name of the assembly on purpose to
+// ensure that we can handle this (mostly an issue for C++ code generation).
+namespace ComWrappersTests
+{
+    internal class Program
+    {
+        public static int Main(string[] args)
```

Comment:
This (along with a `using System.Diagnostics.CodeAnalysis;` at the top) will let you avoid RD.XML. It's the preferred way to do rooting in cases like this (one of the advantages is that this is already documented and in "final form", the other advantage that we don't use here is that the dependency becomes conditional - only if the method attributed with this is compiled, we pull in the dependency).

```suggestion
        [DynamicDependency(DynamicallyAccessedMemberTypes.PublicMethods, typeof(IComInterface))]
        public static int Main(string[] args)
```